corehq/apps/userreports/views.py

```diff
@@ -1216,18 +1220,29 @@ def data_source_status(request, domain, config_id):
     return json_response({'isBuilt': config.meta.build.finished})
 
 
-@login_and_domain_required
-def choice_list_api(request, domain, report_id, filter_id):
+@quickcache(['domain', 'report_id', 'filter_id'], timeout=5 * 60)
```

Comment:
@proteusvacuum Calls (some calls?) to this function fail because `corehq.apps.reports_core.filters.DynamicChoiceListFilter` is not pickleable. I'll offline with you, just wanted to mark the spot.